### PR TITLE
Bug 1182341: Search UI, Identify search field with a magnifying icon at start of field

### PR DESCRIPTION
**Pull request by cindyyu**
Please let me know if I have missed something; this is my first time contributing. 

Thanks!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -38,3 +38,4 @@ media/js/libs/ckeditor/source/ckbuilder
 .tox
 .env
 *_devmo.sql*
+.idea/*
```
Comment (jezdez) [inline .gitignore]: Please don't add such IDE specific items to the repo's gitignore. Instead I would recommend using a [global gitignore](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore).

**Comment by jezdez** [comment]
Thanks for the PR, @cindyyu! I think it's best to wait for @stephaniehobson, @darkwing or @openjck to review it though.

**Comment by stephaniehobson** [comment]
Ours is a complicated code base to be adding to. I'll make a few notes on some line numbers. Thanks for the contribution :)

File: media/stylus/components/home/home-masthead.styl [hunk 2]
```diff
@@ -25,11 +29,17 @@ homepage masthead (search, feature callouts)
...
+    		position: absolute;
+    		top: ($grid-spacing * 2 + 7);
+    }
+
     input#home-q {
         set-placeholder-style(letter-spacing, -1px);
         set-placeholder-style(color, $text-color);
         color: $text-color;
         background-color: #fff;
+        padding-left: 30px;
```
Comment (stephaniehobson) [inline media/stylus/components/home/home-masthead.styl]: We have wiki article in languages that go right to left so we have to code for text going in that direction as well. There's a stylus mixin to simplify that: https://github.com/mozilla/kuma/blob/58dae9a95be2f5fc1bfe4ed377baff0fbdc51c2a/media/stylus/includes/mixins.styl#L316-L346 an example usage would be:
`bidi-style(padding-left, 30px, padding-right, 0)`

And you can test by using the dev tools to change the `<html dir="ltr">` to `rtl`

File: media/stylus/search-suggestions.styl [hunk 3]
```diff
@@ -15,10 +15,15 @@
         clearfix();
     }
 
+    .search-icon {
+    		left: 0;
```
Comment (stephaniehobson) [inline media/stylus/search-suggestions.styl]: Needs to account for RTL display, like the padding above.

File: media/stylus/search-suggestions.styl [hunk 4]
```diff
@@ -15,10 +15,15 @@
...
+    		top: 8px;
+    }
+
     input#home-q {
         border: none;
         display: inline-block;
-        padding: 3px 0;
+        padding: 3px 0 3px 25px;
```
Comment (stephaniehobson) [inline media/stylus/search-suggestions.styl]: Needs to account for RTL display, like the padding above.

**Comment by stephaniehobson** [comment]
The search bar on the search results goes all the way across the screen now when it was 60% wide before. 

screen shot 2015-08-17 at 14 28 57

You seem pretty good with CSS so I'll leave you to fix it but let me know if you need suggestions or if the search results page doesn't display for you locally (I can't remember whether or not search works on a few set up).

**Comment by cindyyu** [comment]
Thank you @stephaniehobson and @jezdez for your feedback! I have incorporated both into my latest commit. 

Please let me know what you think. 

Thanks!

**Comment by stephaniehobson** [comment]
The magnifier over-laps the search filters when they're added on the home page (sorry, I should have seen that when I tested before).

screen shot 2015-08-18 at 15 48 40

It's looking good in RTL and on the search results page now :)

**Comment by cindyyu** [comment]
Hi @stephaniehobson , thanks for pointing that out! I've fixed it.

**Comment by stephaniehobson** [comment]
This is looking good. R :+1: Thanks for the contribution :)

I think we're on a merge freeze due to the problems with our staging environment but there's nothing else @cindyyu needs to do.

**Comment by habber** [comment]
Thanks for your work on this @cindyyu !  Is there a way that I could see how this looks or try it out?

**Comment by stephaniehobson** [comment]
@habber I'll let you know when it pushes to staging but MDN doesn't have any demo servers.

**Comment by stephaniehobson** [comment]
@habber It went from staging to prod super quick and is live now.

**Comment by habber** [comment]
Thanks @stephaniehobson and @cindyyuu. Looks good!

**Comment by openjck** [comment]
Yeah, this is fantastic. Thanks so much @cindyyu!

**Pull request merged**